### PR TITLE
Start making the move of the proxy to an object to make testing easier

### PR DESCRIPTION
**Pull request by natnaelg**
Want to make components more object oriented to make it easier to test. Starting with the proxy.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,22 +25,17 @@ import (
 var cpuprofile = flag.String("cpuprofile", "", "write cpu profile to file")
 
 func main() {
+	runtime.GOMAXPROCS(runtime.NumCPU())
```
Comment (natnaelg) [inline main.go]: Adding this for Go backwards compatibility

Comment (heathj) [reply main.go]: I don't think this program runs on Golang version <= 1.10, actually. We used a JSON API that is fairly new in the `encoding/json` https://golang.org/pkg/encoding/json/#Valida and due to a weird compiling error in CGO on Macs, it won't compile on anything less than 1.9 https://github.com/nccgroup/tracy/wiki/Building#golang. So not sure that changes anything here.

**Comment by natnaelg** [comment]
Mmk I'll remove

File: main.go (hunk 1 above)
Comment (heathj) [reply main.go]: https://github.com/nccgroup/tracy/blob/863645e99b0811751aa7aa1c78d4f0ffd8517f79/api/common/event.go#L251

**Comment by natnaelg** [comment]
@jacobheath G2g, tested locally with curl and things look good.

**Pull request merged**